Tutorials/CNTK_209C_Image_Super-resolution_Using_CNNs_and_GANs.ipynb

```diff
@@ -0,0 +1,1562 @@
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<img src=\"https://superresolution.blob.core.windows.net/superresolutionresources/vdsr_architecture.png\"/>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Image object>"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "import numpy as np\n",
+    "from scipy.misc import imsave\n",
+    "from IPython.display import Image\n",
+    "import urllib\n",
+    "try:\n",
+    "    from urllib.request import urlretrieve, urlopen\n",
+    "except ImportError: \n",
+    "    from urllib import urlretrieve, urlopen\n",
+    "\n",
+    "import cntk as C\n",
+    "from cntk import Trainer\n",
+    "from cntk import load_model, combine\n",
+    "from cntk.initializer import he_normal\n",
+    "from cntk.layers import BatchNormalization, Convolution, ConvolutionTranspose2D, Dense\n",
+    "from cntk.ops import relu, leaky_relu, reduce_mean, square, param_relu\n",
+    "from cntk.io import (MinibatchSource, ImageDeserializer, StreamDef, StreamDefs)\n",
+    "from cntk.learners import (adam, UnitType, learning_rate_schedule, momentum_schedule)\n",
+    "import cntk.io.transforms as xforms\n",
+    "\n",
+    "#Figure 1\n",
+    "Image(url = \"https://superresolution.blob.core.windows.net/superresolutionresources/vdsr_architecture.png\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "<b>Figure 1</b> above was taken from the paper <a href=\"https://arxiv.org/pdf/1511.04587.pdf\">Accurate Image Super-Resolution Using Very Deep Convolutional Networks</a>. It shows the architecture of <b>VDSR</b> model. We start from the interpolated low resolution image (ILR) and predict the residual image. Most values in it are zero or very small. Adding the residual image and ILR gives us our prediction of high resolution image."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "### Select the notebook runtime environment devices / settings\n",
+    "Set the device to cpu / gpu for the test environment. If you have both CPU and GPU on your machine, you can optionally switch the devices. By default, we choose the best available device."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Select the right target device when this notebook is being tested:\n",
+    "if 'TEST_DEVICE' in os.environ:\n",
```

Comment:
This notebook should be tested and used only on GPU (look at test file). This is because batch normalization training is not yet implemented on CPU. How should this part look, then? I will also have to add a comment explaining this in the notebook.